### DISCUSSION TITLE
How to inspect flood process?

### DISCUSSION THREAD

**Discussion by trim21** [jesec/flood#420] [Q&A]
I'm using deb package un my Debian NAS, but the memory usage increased from 100MB to 5GB.

I think there is a memory leak and I try to debug it with `NODE_OPTIONS="--inspect=0.0.0.0:9229"` but it looks that `flood: --inspect= is not allowed in NODE_OPTIONS`

Is there any way I can run it from by flood bin? Or I must clone the code and run it with node?

**Comment by jesec** [comment]
You have to. The single-executable binary does not allow debugging.

**Reply by trim21** [reply]
there is definitely a memory leak...

![image](https://github.com/jesec/flood/assets/13553903/08584e39-957b-4ff8-b0bb-3a62f3ee99e1)


Now I just need to find the cause...

**Reply by trim21** [reply]
it's caused by `TORRENT_LIST_DIFF_CHANGE`
![image](https://github.com/jesec/flood/assets/13553903/7cca4f6c-d123-4a56-a214-506d226324f3)

**Reply by trim21** [reply]
some are caused by rtorrent `methodCallJSON`
